### DISCUSSION TITLE
Add copy button to code snippets

### DISCUSSION THREAD

**Discussion by Complexlity** [TheOdinProject/theodinproject#3768] [Ideas]
Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this issue follows the `brief description of request` format, e.g. `Add dark mode to website`

The following checkbox is OPTIONAL:

-   [ ] I would like to be assigned this issue to work on it

<hr>

**1. Description of the Feature Request:**

Similar to the universal style to coding blocks, I would be an improvement if there's a copy button on code blocks to easily get the entire code without having to highlight it all (especially for long code blocks)

**2. Acceptance Criteria:**

-[x] A simple copy code button

**3. Additional Information:**

![image](https://user-images.githubusercontent.com/105590967/235274112-1ee36aed-a1a1-4b6b-8c9c-f93ebfca02f7.png)
An example of it

**Comment by KevinMulhern** [comment]
Hey @Complexlity, thanks for the suggestion.

I can see something like being useful for installation snippets, we tell learners to copy and paste those anyway to avoid issues. 

But I don't think we should enable this for code examples.  From a learning perspective, theres a lot to be gained from manually typing code examples out:

* Building muscle memory with new syntax. 
* Opportunities to understand and play with new concepts as you type it out.
* Getting used to debugging when typos are inevitably made.

I'd definitely like more opinions on this though.

**Comment by StuvaScript** [comment]
If anything, the text highlight should be turned off so that nobody can copy and paste. One thing that new developers don't realize is the huge benefit to typing out your code. Especially in the age of AI, people get so reliant on copy and paste that when asked to type any code it becomes harder to draw it from memory without putting in the reps of typing out your code. 

I'm not saying absolutely no copy and pasting allowed but beginner devs should definitely be getting as much code typing time as possible. 👽
